### PR TITLE
Normalization support destination sync modes append_dedup #2372

### LINKED ISSUE #2372: Normalization support destination sync modes
## Tell us about the problem you're trying to solve
As part of #2337, we would like to introduce a new configuration enum for sync modes on the destination side.
Depending on the configuration of the AirbyteCatalog in #2370, the normalization should generate additional SQL models for DBT to transform and dedupe data from a stream (SCD type 1 or 2).

## Describe the solution you’d like
DBT pipeline reflecting the configuration of the catalog for each stream.

## Additional context
https://docs.google.com/document/d/1uHGz-soEdGcAe4Mgf0nqUpst6NszkTGL5aJivJ4qxpM/edit#

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
*Describe what the change is solving*
Closes #2372 

## How
*Describe the solution*
Add extra models steps in dbt pipeline to dedup

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/catalog_processor.py [hunk 1]
```diff
@@ -124,6 +140,10 @@ def build_stream_processor(
                 destination_type=destination_type,
                 raw_schema=raw_schema_name,
                 schema=schema_name,
+                source_sync_mode=source_sync_mode,
```
Comment (cgardens) [inline airbyte-integrations/bases/base-normalization/normalization/transform_catalog/catalog_processor.py:143]: i'm having trouble understanding where source_sync_mode is used after it is passed in there. am i just missing it or is it a future thing?

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py [hunk 2]
```diff
@@ -460,13 +590,15 @@ def generate_new_table_name(self, is_intermediate: bool, suffix: str) -> str:
         tables_registry = self.tables_registry.union(self.local_registry)
         new_table_name = self.normalized_stream_name()
         if not is_intermediate and self.parent is None:
+            if suffix:
```
Comment (cgardens) [inline airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py:593]: what problem is this change solving? are we going to end up with users_ab1_properties_ab1 or something funky like that? 

maybe there's a unit test we could add to make it clear what the expected behavior should be here. to be clear, i don't have an opinion on what it should be, i just am struggling to understand what it is based on the code. i think unit tests will help document it.

**Review by cgardens — APPROVED** [review]

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/catalog_processor.py (hunk 1 above)
Comment (ChristopheDuong) [reply airbyte-integrations/bases/base-normalization/normalization/transform_catalog/catalog_processor.py:143]: yes you're right it's not currently used, this Is just registering what the source sync was and can be used in the future on the normalization side

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py (hunk 2 above)
Comment (ChristopheDuong) [reply airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py:593]: The suffix was used only for intermediate tables so far (tables ending with (_ab1, _ab2, etc)

In this PR, I added an extra non-intermediate table (materialized and exposed to the user) with a suffix `_scd` which is the history SCD type 2 table in addition to the usual table per stream (which is deduped SCD type 1)

So this change makes it possible to suffix non-intermediate tables.

**Pull request merged**